src/main/java/Duke.java

```diff
@@ -1,10 +1,91 @@
+import java.util.Scanner;
+import java.util.Arrays;
+
 public class Duke {
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        final String DIVIDER = "    ——————————————————————————————————————————————————————————————";
+        System.out.println(DIVIDER);
+        System.out.println("    Hello! I'm Duke\n    What can I do for you?");
+        System.out.println(DIVIDER);
+        String line;
+        String[] lists = new String[100];
+        String[] flags = new String[100];
+        String[] doneTasks = new String[100];
+        String[] dates = new String[100];
+        Arrays.fill(doneTasks," ");
+        int number = 0;
+        Scanner in = new Scanner(System.in);
+        line = in.nextLine();
+        while (!line.equals("bye")) {
+            if (line.equals("list")) {
+                System.out.println(DIVIDER);
+                System.out.println("    " + "Here are the tasks in your list:");
+                for(int i = 0; i < number; i++) {
+                    System.out.print("    " + (i + 1) + ". " + "[" + flags[i] + "]"
+                            + "[" + doneTasks[i] + "] " + lists[i]);
+                    if(flags[i].equals("D")) {
+                        System.out.println(" (by: " + dates[i] + ")");
+                    } else if (flags[i].equals("E")) {
+                        System.out.println(" (at: " + dates[i] + ")");
+                    } else {
+                        System.out.println(" ");
+                    }
+                }
+                System.out.println(DIVIDER);
+                line = in.nextLine();
+            }  else if (line.contains("done")) {
+                int taskDone = Integer.parseInt(line.substring(5));
+                System.out.println(DIVIDER);
+                System.out.println("    Nice! I've marked this task as done:");
+                doneTasks[taskDone - 1] = "X";
+                System.out.print("      [" + flags[taskDone - 1] + "][X] " + lists[taskDone - 1]);
+                if(flags[taskDone - 1].equals("D")) {
```

Comment:
Create a separate function for these if-else statements to prevent over-nesting.